.github/workflows/ci.yml

```diff
@@ -1087,3 +1087,35 @@ jobs:
         with:
           webhook: ${{ secrets.DISCORD_WEBHOOK }}
           description: "The following jobs failed: ${{ steps.failed_jobs.outputs.failed-jobs }}"
+      - name: Show useful artifact links
+        if: always()
+        env:
+          BUILD_DIR_GCS_ARTIFACT: ${{ needs.build_all.outputs.build-dir-gcs-artifact }}
+          E2E_TEST_ARTIFACTS_GCS_ARTIFACT_DIR: ${{ needs.build_e2e_test_artifacts.outputs.e2e-test-artifacts-gcs-artifact-dir }}
+          BENCHMARK_TOOLS_GCS_ARTIFACT_DIR: ${{ needs.build_benchmark_tools.outputs.benchmark-tools-gcs-artifact-dir }}
+          EXECUTION_BENCHMARK_RESULTS_GCS_ARTIFACT_DIR: ${{ needs.execution_benchmarks.outputs.benchmark-results-gcs-artifact-dir }}
+          COMPILATION_BENCHMARK_RESULTS_GCS_ARTIFACT: ${{ needs.compilation_benchmarks.outputs.compile-stats-results-gcs-artifact }}
+        # If the job of an artifact is skipped or failed, we show "Not present".
+        run: |
+          cat <<EOF >> "${GITHUB_STEP_SUMMARY}"
+          ### :link: Artifact Links
+
+          \`\`\`sh
+          # Compiler and host tools archive:
+          export BUILD_DIR_URL="${BUILD_DIR_GCS_ARTIFACT:-Not present}"
+          # Directory of e2e test artifacts:
+          export E2E_TEST_ARTIFACTS_DIR_URL="${E2E_TEST_ARTIFACTS_GCS_ARTIFACT_DIR:-Not present}"
+          # Directory of benchmark tool binaries:
+          export BENCHMARK_TOOLS_DIR_URL="${BENCHMARK_TOOLS_GCS_ARTIFACT_DIR:-Not present}"
+          # Directory of execution benchmark results and traces:
+          export EXECUTION_BENCHMARK_RESULTS_DIR_URL="${EXECUTION_BENCHMARK_RESULTS_GCS_ARTIFACT_DIR:-Not present}"
+          # Compilation benchmark results:
+          export COMPILATION_BENCHMARK_RESULTS_URL="${COMPILATION_BENCHMARK_RESULTS_GCS_ARTIFACT:-Not present}"
+          \`\`\`
+
+          You can list \`gs://\` directories and download files with [gcloud cli](
+          https://cloud.google.com/sdk/gcloud/reference/storage).
+
+          To run benchmarks locally with the CI-built e2e test artifacts, see [IREE Benchmark Suites](
+          https://github.com/${GITHUB_REPOSITORY}/blob/main/docs/developers/developing_iree/benchmark_suites.md).
```

Comment:
Can you link to this section directly?
https://github.com/openxla/iree/blob/main/docs/developers/developing_iree/benchmark_suites.md#3-fetch-the-benchmark-artifacts

The "links" above just set some environment variables, so knowing how to use those variables is important. You could also include a few frequently used commands fully ready to copy-paste in a single step, like
```bash
mkdir -p "/tmp/iree-artifacts"

# Download all artifacts
gcloud storage cp -r "${E2E_TEST_ARTIFACTS_GCS_ARTIFACT_DIR?}" "/tmp/iree-artifacts"
```